Libs/MRML/Core/vtkMRMLSliceNode.cxx

```diff
@@ -371,73 +408,194 @@ void vtkMRMLSliceNode::SetOrientationToReformat()
-    this->SliceToRAS->SetElement(0, 2,  0.0);
-    this->SliceToRAS->SetElement(1, 2,  1.0);
-    this->SliceToRAS->SetElement(2, 2,  0.0);
-
-    this->SetOrientationString( "Coronal" );
-    this->SetOrientationReference( "Coronal" );
-    this->UpdateMatrices();
+  this->SliceToRAS->DeepCopy(this->GetSliceOrientationPreset("Coronal"));
+
+  //SetOrientationString() and SetOrientationReference()
+  //behaviour will be mantained for backward compatibility
+  this->SetOrientationString("Coronal");
+  this->SetOrientationReference("Coronal");
+  this->UpdateMatrices();
 }
 
 //----------------------------------------------------------------------------
 // Local helper to compare matrices -- TODO: is there a standard version of this?
 bool vtkMRMLSliceNode::Matrix4x4AreEqual(const vtkMatrix4x4 *m1, const vtkMatrix4x4 *m2)
 {
-  return vtkAddonMathUtilities::Matrix4x4AreEqual(m1, m2, /* tolerance= */ 0);
+  return vtkAddonMathUtilities::Matrix4x4AreEqual(m1, m2);
+}
+
+//----------------------------------------------------------------------------
+vtkMatrix4x4 *vtkMRMLSliceNode::GetSliceOrientationPreset(const std::string &name)
+{
+  if (!name.compare("Reformat"))
+    {
+    return NULL;
+    }
+
+  std::map<vtkStdString, vtkMatrix4x4*>::iterator it = this->OrientationMatrices.find(name);
+  if (it != this->OrientationMatrices.end())
+    {
+    return it->second;
+    }
+
+  vtkErrorMacro("GetSliceOrientationPreset: invalid Orientation Name: " << name);
+  return NULL;
+}
+
+//----------------------------------------------------------------------------
+std::string vtkMRMLSliceNode::GetSliceOrientationPresetName(vtkMatrix4x4 *sliceToRAS)
+{
+  if (sliceToRAS == NULL)
+    {
+    vtkErrorMacro("GetSliceOrientationPresetName: invalid input Matrix. ");
+    return "";
+    }
+
+  std::map<vtkStdString, vtkMatrix4x4*>::iterator it;
+  for (it = this->OrientationMatrices.begin(); it != this->OrientationMatrices.end(); ++it)
+    {
+    std::string orietation = it->first;
+    if (!orietation.compare("Reformat"))
+      {
+      continue;
+      }
+    vtkMatrix4x4* storedSliceToRAS = this->GetSliceOrientationPreset(orietation);
+    if (storedSliceToRAS != NULL)
+      {
+      if (vtkMRMLSliceNode::Matrix4x4AreEqual(storedSliceToRAS, sliceToRAS))
+        {
+        return orietation;
+        }
+      }
+    }
+
+  vtkErrorMacro("GetSliceOrientationPresetName: orientation string not found.");
+  return "";
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLSliceNode::GetSliceOrientationPresetNames(vtkStringArray *namedOrientations)
+{
+  namedOrientations->SetNumberOfValues(this->OrientationMatrices.size() + 1);
+
+  std::map<vtkStdString, vtkMatrix4x4*>::iterator it;
+  int id = 0;
+  for (it = this->OrientationMatrices.begin(); it != this->OrientationMatrices.end(); ++it)
+    {
+    namedOrientations->SetValue(id, it->first);
+    id++;
+    }
+  namedOrientations->SetValue(id, "Reformat");
```

Comment:
I don't think it's a good idea to return the "Reformat" string along with the real orientation preset names. The "Reformat" option can be added at the GUI level (e.g., the widget asks for the orientation preset names and it may add the "Reformat" name if it wants to).